classes/local/store/azure_blob_storage/client.php

```diff
@@ -0,0 +1,296 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace tool_objectfs\local\store\azure_blob_storage;
+
+use admin_settingpage;
+use coding_exception;
+use GuzzleHttp\Psr7\Utils;
+use tool_objectfs\local\store\object_client_base;
+use local_azureblobstorage\api;
+use local_azureblobstorage\stream_wrapper;
+use stdClass;
+use Throwable;
+
+/**
+ * Azure blob storage client
+ *
+ * @package    tool_objectfs
+ * @author     Matthew Hilton <matthewhilton@catalyst-au.net>
+ * @copyright  2024 Catalyst IT
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class client extends object_client_base {
+
+    /** @var api $api Azure API */
+    protected api $api;
+
+    /**
+     * Creates object client
+     * @param stdClass $config / TODO is this maybe null ?
+     */
+    public function __construct($config) {
+        if (empty($config) || !$this->get_availability()) {
+            parent::__construct($config);
+            return;
+        }
+
+        $this->api = new api($config->azure_accountname, $config->azure_container, $config->azure_sastoken);
+        $this->config = $config;
+        $this->maxupload = api::MAX_BLOCK_SIZE;
+    }
+
+    /**
+     * Determines if this filesystem is available for use.
+     * @return bool
+     */
+    public function get_availability(): bool {
+        // Requires local_azureblobstorage to be installed.
+        // Namespace changed in 4.4+.
+        if (class_exists('\core\plugin_manager')) {
+            $info = \core\plugin_manager::instance()->get_plugin_info('local_azureblobstorage');
+            // For 4.2 and 4.3.
+        } else if (class_exists('\core_plugin_manager')) {
+            $info = \core_plugin_manager::instance()->get_plugin_info('local_azureblobstorage');
+        } else {
+            throw new coding_exception("Could not load plugin manager class");
+        }
+
+        // Info is empty if plugin is not installed or no API is setup (missing config?).
+        return !empty($info);
+    }
+
+    /**
+     * Sets the StreamWrapper to allow accessing the remote content via a blob:// path.
+     */
+    public function register_stream_wrapper() {
+        if ($this->get_availability()) {
+            stream_wrapper::register($this->api);
+        } else {
+            parent::register_stream_wrapper();
```

Comment:
This is just void in the parent, perhaps we just early return instead here, its a bit clearer that this actually does nothing